### PR TITLE
feat(catalog): add Link to tags and other fields in AboutCard

### LINKED ISSUE #24213: 🚀 Feature: Making the tags clickable in the AboutCard Component
### 🔖 Feature description

The tags in the AbouCard component should be clickable to make it easier to filter the entities by tags.

Screen Shot 2024-04-12 at 14 28 20

### 🎤 Context

Most users expect the chip component used for tags to be clickable, and expect it to navigate to the catalog list page with a preselected filter for the tag.

### ✌️ Possible Implementation

The tags can point to the catalog page with `?filters[tags]=${tag}` search parameter.

I believe this line would have to be changed/enclosed within a `Link` tag: https://github.com/backstage/backstage/blob/master/plugins/catalog/src/components/AboutCard/AboutContent.tsx#L202

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by deshmukhmayur**
## Hey, I just made a Pull Request!

Added a Link component to the tags, lifecycle and type fields in the `AboutContent` inside `AboutCard`.

This makes the fields clickable and navigates the user to the catalog index page, with preselected filters.

Resolves #24213

Screen Shot 2024-04-22 at 18 55 27

image

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by riginoommen — APPROVED** [review]

**Comment by deshmukhmayur** [comment]
My bad. I think this should probably be a minor version bump, not a patch. Updated the changeset again.

**Comment by deshmukhmayur** [comment]
Hello, any update on this?
cc. @awanlin

File: plugins/catalog/src/components/AboutCard/AboutContent.tsx
Comment (awanlin) [inline plugins/catalog/src/components/AboutCard/AboutContent.tsx:106]: Best I can tell there is nothing that's actually testing this function, can we make sure to add tests for this, please?

File: yarn.lock
Comment (awanlin) [inline yarn.lock:12618]: Based on your changes there should be no changes to this file, can you please revert?

**Review by awanlin — CHANGES_REQUESTED** [review]
Thanks for the contribution @deshmukhmayur, over all this looks good but I'm deferring final say to the Maintainers. I did leave two comments though

File: yarn.lock
Comment (deshmukhmayur) [reply yarn.lock]: Removed it from the commit.

But I see that the file is being changed whenever I run yarn install, not sure why though.

File: plugins/catalog/src/components/AboutCard/AboutContent.tsx
Comment (deshmukhmayur) [reply plugins/catalog/src/components/AboutCard/AboutContent.tsx:106]: I see the existing unit tests only test for text content, so I didn't make any changes to that.

I'll try to add tests for the links.

**Comment by benjdlambert** [comment]
Ping @deshmukhmayur did you manage to get round to having a look at adding some tests? :pray:

**Comment by deshmukhmayur** [comment]
Hey @benjdlambert , I got busy with other work so haven't had the chance to add the tests yet. Will work on it as soon as I get some time by next couple of weeks.

**Comment by awanlin** [comment]
Hi @deshmukhmayur, any updates on your end? This seems very close to being something we can ship just needs the tests.

File: plugins/catalog/src/components/AboutCard/AboutContent.tsx
Comment (camilaibs) [inline plugins/catalog/src/components/AboutCard/AboutContent.tsx:228]:
```suggestion
            component={Link}
            to={getPathForFilter('tags', t)}
```

**Review by camilaibs — CHANGES_REQUESTED** [review]
This looks amazing, thanks for the contribution ⭐ 
Left just minor comments 🙂

**Pull request closed without merging**
